WinPort/src/Backend/TTY/TTYX/TTYX.cpp

```diff
@@ -155,7 +155,26 @@ class TTYX
 			if ((cookie->evtype == XI_RawKeyPress || cookie->evtype == XI_RawKeyRelease) && cookie->data) {
 				const XIRawEvent *ev = (const XIRawEvent *)cookie->data;
 				//fprintf(stderr, "TTYXi: !!!!! %d %d\n", cookie->evtype == XI_RawKeyPress, ev->detail);
-				const KeySym ks = XkbKeycodeToKeysym(_display, ev->detail, 0 /*xkbState.group*/, 0 /*shift level*/);
+
+				XkbDescPtr xkb = XkbGetMap(_display, 0, XkbUseCoreKbd);
```

Comment:
its better to call XkbGetMap/XkbGetControls/XkbGetNames in constructor of  TTYX remembering all needed things and do here only XkbKeycodeToKeysym

Reply:
If done this way, there may be errors if the user adds or removes keyboard layouts while far2l is already running.

Reply:
I think its a very rare situation and user will restart far2l before posting bug here)
But once somebody will post a bug, will then use https://tronche.com/gui/x/xlib/events/window-state-change/mapping.html

Reply:
ok, fixed!

Reply:
Is there something else left to be done?